### ISSUE TITLE
Players incorrectly being reported as observers in BWAPI 4.2.0 clients

### ISSUE THREAD

**Issue by Jabbo16** [OpenBW/BWAPI4J#38]
This code called at `onStart()` while running a Melee game mode match:
```java
for(Player p : bw.getAllPlayers()) {
    System.out.println("playername: " + p.getName() + " is observer: " + p.isObserver());
}
```

Returns all players as being observers:

```
playername: Cronus Wing is observer: true
playername: bwapi is observer: true
playername: Neutral is observer: true
```

The only `Player` that should be observer is `Neutral`.

**Comment by adakitesystems** [comment]
Confirmed. However, in Linux + OpenBW + BWAPI4J, these players are not reported as observers. I suspect something to do with the bridge for Windows. Looking into recompiling the bridge. The values seem to be getting set correctly:
````
intBuf[index++] = player->isObserver() ? 1 : 0;
````
and
````
this.isObserver = playerData[index + Player.IS_OBSERVER_INDEX] == 1;
````

**Comment by adakitesystems** [comment]
Update:
- Recompiling the bridge for Windows did not resolve the issue.
- As a manual debug test, changing `intBuf[index++] = player->isObserver() ? 1 : 0;` to `intBuf[index++] = player->isObserver() ? 0 : 0;` in the Windows bridge resulted in the players being correctly reported as `isObserver() == false`.
- Somehow, the Windows bridge is reporting the incorrect value but the Linux bridge appears to report the correct value.

**Comment by Jabbo16** [comment]
Shouldnt the code be:

`intBuf[index++] = player->isObserver() == 1 ? 1 : 0`

C++ Api `isObserver()` method return `1` if observer and '0` if not.

**Comment by adakitesystems** [comment]
`PlayerInterface.isObserver()` in C++ is a bool function. That is the correct syntax. Even if it wasn't a bool, that is still valid syntax. I tried with `== 1` anyway and observed the same incorrect behavior.
![bool](https://user-images.githubusercontent.com/22711643/39960580-83b2ef08-55e2-11e8-971c-3285bddef663.png)

**Comment by Jabbo16** [comment]
Ok, then forget what I said.

What a weird behaviour.

**Comment by adakitesystems** [comment]
Using a completely separate C++ BWAPI 4.2.0 client project with no Java, I am getting the same incorrect behavior with the client version but not with the module version. @Jabbo16, can you verify this as well, please?

**Comment by Jabbo16** [comment]
I checked module before and got correct behaviour, both players reported as not observers, just the Neutral player as observer.

**Comment by adakitesystems** [comment]
Yes, it looks like a BWAPI client bug that is present in 4.2.0, 4.1.2, and maybe other versions. The module appears unaffected. If/when it gets fixed, someone (if I'm not around) should use the updated BWAPI libs to recompile the bridge and then close this issue.

**Comment by adakitesystems** [comment]
This issue has been fixed in the `develop` branch of C++ BWAPI. When an official version is released with this fix, the BWAPI4JBridge just needs to update to the new BWAPI version and recompile the bridge. Then, this issue can be closed.

**Comment by adakitesystems** [comment]
Upgraded to BWAPI v4.4.0 in https://github.com/OpenBW/BWAPI4J/commit/83423a9113bd382c71e35690936e8c62142a2206 which should resolve this. Needs to be tested first.
